ports/mimxrt/mpconfigport.h

```diff
@@ -177,21 +191,44 @@ extern const struct _mp_obj_type_t network_lan_type;
 #define MICROPY_HW_NIC_ETH
 #endif
 
+#if MICROPY_PY_NETWORK_NINAW10
+// This Network interface requires the extended socket state.
+#ifndef MICROPY_PY_SOCKET_EXTENDED_STATE
+#define MICROPY_PY_SOCKET_EXTENDED_STATE    (1)
+#endif
+extern const struct _mp_obj_type_t mod_network_nic_type_nina;
+#define MICROPY_HW_NIC_NINAW10              { MP_ROM_QSTR(MP_QSTR_WLAN), MP_ROM_PTR(&mod_network_nic_type_nina) },
+
+#else
+
+#define MICROPY_HW_NIC_NINAW10
+
+#endif // MICROPY_PY_NETWORK_NINAW10
+
 #if MICROPY_PY_NETWORK_CYW43
 extern const struct _mp_obj_type_t mp_network_cyw43_type;
 #define MICROPY_HW_NIC_CYW43                { MP_ROM_QSTR(MP_QSTR_WLAN), MP_ROM_PTR(&mp_network_cyw43_type) },
 #else
 #define MICROPY_HW_NIC_CYW43
 #endif
 
+#if MICROPY_PY_NETWORK_ESP_HOSTED
+extern const struct _mp_obj_type_t mod_network_esp_hosted_type;
+#define MICROPY_HW_NIC_ESP_HOSTED   { MP_ROM_QSTR(MP_QSTR_WLAN), MP_ROM_PTR(&mod_network_esp_hosted_type) },
+#else
+#define MICROPY_HW_NIC_ESP_HOSTED
+#endif
```

Comment:
I would use `MICROPY_BOARD_NETWORK_INTERFACES` to add the new interfaces.

Reply:
But that would required to add it to every supported board, instead making it depending on MICROPY_PY_NETWORK_ESP_HOSTED